bluesky/run_engine.py

```diff
@@ -270,7 +270,7 @@ def _clear_run_cache(self):
         self._metadata_per_run.clear()
         self._bundling = False
         self._run_is_open = False
-        self._msg_cache = None  # checkpoints can't rewind into a closed run
```

Comment:
In effect we should have `self._run_is_open == self._run_start_uid is not None` at all times.